From f0088f12e7ddc2bf7de1e14c516ca9e66dc2f6b5 Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Mon, 27 Jul 2020 14:10:06 +0900
Subject: [PATCH 44/63] BugTrackingSystem

650 - Client redirection URL does not apply Trust/Untrust judgement module.
---
 content/browser/web_contents/web_contents_impl.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/content/browser/web_contents/web_contents_impl.cc b/content/browser/web_contents/web_contents_impl.cc
index 4460704f22296..440633e27e437 100644
--- a/content/browser/web_contents/web_contents_impl.cc
+++ b/content/browser/web_contents/web_contents_impl.cc
@@ -5656,7 +5656,9 @@ void WebContentsImpl::DidFinishNavigation(NavigationHandle* navigation_handle) {
 
 #if BUILDFLAG(ENABLE_GOOROOM)
 //If browser is switched, the current page is closed.
-  if (navigation_handle->IsInMainFrame() && RequestCurrentUserDataDir(navigation_handle->GetURL())) {
+  if (navigation_handle->IsInMainFrame()
+      && (navigation_handle->GetPageTransition() & ui::PAGE_TRANSITION_CLIENT_REDIRECT) == 0
+      && RequestCurrentUserDataDir(navigation_handle->GetURL())) {
     ClosePage();
   }
 #endif
-- 
2.30.2

